programs/management/commands/import-projection-data.py

```diff
@@ -0,0 +1,168 @@
+from django.core.management.base import BaseCommand, CommandError
+from programs.models import *
+
+import decimal
+import urllib2
+import itertools
+import logging
+import sys
+import csv
+import ssl
+
+
+class Command(BaseCommand):
+    help = 'Imports Bureau of Labor Statistics Employment projections'
+
+    projection_data = []
+    projections_added = 0
+    projections_skipped = 0
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'file',
+            type=file,
+            help='The file path of the csv file'
+        )
+        parser.add_argument(
+            '--verbose',
+            help='Use verbose logging',
+            action='store_const',
+            dest='loglevel',
+            const=logging.INFO,
+            default=logging.WARNING,
+            required=False
+        )
+        parser.add_argument(
+            '--cip-version',
+            type=str,
+            dest='cip_version',
+            help='The version of CIPs used by programs in this import.',
+            default=settings.CIP_CURRENT_VERSION,
+            required=False
+        )
+        parser.add_argument(
+            '--soc-version',
+            type=str,
+            dest='soc_version',
+            help='The version of the SOC specification to set.',
+            default=settings.SOC_CURRENT_VERSION,
+            required=False
+        )
+        parser.add_argument(
+            '--report-version',
+            type=str,
+            dest='report_version',
+            help='The version of the report being imported. Usually a range of years.',
+            default=settings.PROJ_CURRENT_REPORT,
+            required=False
+        )
+
+    def handle(self, *args, **options):
+        self.file = options['file']
+        self.loglevel = options['loglevel']
+        self.cip_version = options['cip_version']
+        self.soc_version = options['soc_version']
+        self.report_version = options['report_version']
+
+        # Set logging level
+        logging.basicConfig(stream=sys.stdout, level=self.loglevel)
+
+        # Fetch all outcome data:
+        self.get_projections(self.file)
+
+        # Assign outcome data to existing programs:
+        self.assign_projections()
+
+        # Print results
+        self.print_results()
+
+    def get_projections(self, csv_file):
+
+        try:
+            csv_reader = csv.DictReader(csv_file)
+        except csv.Error:
+            logging.error('\n ERROR reading CSV: CSV does not have valid headers or is malformed.')
+            return
+
+        for row in csv_reader:
```

Comment:
You may want to consider adding some sort of filetype sniffing here before attempting to loop through each CSV row; in my experience, the csv reader lib will happily loop through any non-CSV document if provided